pkg/cloud/services/ec2/instances.go

```diff
@@ -262,26 +262,54 @@ func (s *Service) findSubnet(scope *scope.MachineScope) (string, error) {
 
 	switch {
 	case scope.AWSMachine.Spec.Subnet != nil && scope.AWSMachine.Spec.Subnet.ID != nil:
+		if failureDomain != nil {
+			subnets, err := s.getFilteredSubnets(filter.EC2.SubnetID(aws.StringValue(scope.AWSMachine.Spec.Subnet.ID)))
+			if err != nil {
+				return "", err
```

Comment:
Could potentially wrap the error, but a nice to have

Reply:
Fixed it, Thanks!